src/components/pages/AlphabeticComparisonPage/SearchBox.tsx

```diff
@@ -1,122 +1,251 @@
-import React, { Component } from "react";
-import { Innholdstype, MainElement } from "../../../data/ApiTypes";
+import React, { Component, Fragment } from "react";
+import { Innholdstype, Suggest, SuggestElement } from "../../../data/ApiTypes";
 import { API_DOMAIN } from "../../../data/config";
 import { with_app_state, AppStateProps } from "../../app/AppContext";
 import styles from "./SearchBox.module.scss";
 import { ReactComponent as Search } from "../../../fontawesome/solid/search.svg";
 import { objectToQueryString } from "../../../util/querystring";
+import Translate from "../../app/Translate";
+import { Redirect } from "react-router";
 
 type Props = {
-  innholdstype: Innholdstype;
+  innholdstype?: Innholdstype;
 };
 
 type State = {
-  data: MainElement[];
+  suggestions: { [type: string]: SuggestElement[] };
+  numSuggestions: number;
   searchString: string;
+  activeSuggestion: number;
   error: boolean;
-  listText: string;
+  redirect: boolean;
+};
+
+const Innholdstyper: { [type: string]: JSX.Element } = {
+  u: <Translate nb="Utdanninger" />,
+  y: <Translate nb="Yrker" />,
 };
 
 class SearchBox extends Component<Props & AppStateProps, State> {
-  state = {
-    data: [],
+  state: State = {
+    suggestions: {},
+    numSuggestions: 0,
     searchString: "",
+    activeSuggestion: -1,
     error: false,
-    listText: "",
+    redirect: false,
   };
 
   handleChange = (event: any) => {
+    const { innholdstype } = this.props;
     const value = event.target.value;
 
+    this.setState({
+      suggestions: {},
+      numSuggestions: 0,
+      searchString: value,
+      activeSuggestion: -1,
+    });
     if (value.length < 3) {
-      this.setState({
-        data: [],
-        searchString: value,
-        listText: "Skriv inn minst tre tegn for å søke...",
-      });
       return;
     }
-    this.setState({ searchString: value });
 
-    fetch(API_DOMAIN + "/rest/suggest?" + objectToQueryString({ q: value }))
+    fetch(
+      API_DOMAIN +
+        "/rest/suggest?" +
+        objectToQueryString({
+          q: value,
+          innholdstype:
+            innholdstype == "utdanning"
+              ? "utdanningsbeskrivelse"
```

Comment:
Skal man ikke kunne søke på alle innholdstyper om den ikke er satt? Og få opp liste med yrker OG utdanninger? mulig du gjør det, bare ser at api-kallet her sender med en spesifikk?

Reply:
Joda, helt rett det, og det er slik koden virker (hvis jeg ikke har dummet meg ut).

Reply:
hvis innholdstype ikke er satt her, så sender den ingenting til apiet

Reply:
Men vil vi ikke ha ut data fra APIet, for begge innholdstyper?

Reply:
men hvis innholstype er "utdanning" sender den "utdanningsbeskrivelse". Jeg synes ikke det blir så heldig å bruke "utdanningspbeskrivelse" over alt i url osv fordi apiet krever det

Reply:
når innholdstype ikke er satt sender apiet alle innholdstyper

Reply:
👍